src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,623 @@
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.filters;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import com.puppycrawl.tools.checkstyle.PropertyType;
+import com.puppycrawl.tools.checkstyle.XdocsPropertyType;
+import com.puppycrawl.tools.checkstyle.api.AuditEvent;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.Filter;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+/**
+ * <p>
+ * Filter {@code SuppressWithNearbyTextFilter} uses plain text to suppress
+ * nearby audit events. The filter can suppress all checks which have Checker as a parent module.
+ * </p>
+ * <p>
+ * Rationale: Sometimes there are legitimate reasons for suppressing a check.
+ * When this is a matter of the code in question and not personal preference,
+ * the best place to override the policy is in the code itself. Semi-structured
+ * comments can be associated with the check. This is sometimes superior to
+ * a separate suppressions file, which must be kept up-to-date as the source
+ * file is edited.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code nearbyTextPattern} - Specify nearby text
+ * pattern to trigger filter to begin suppression.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code "SUPPRESS CHECKSTYLE (\w+)"}.
+ * </li>
+ * <li>
+ * Property {@code checkPattern} - Specify check name pattern to suppress.
+ * Property can also be a RegExp group index at {@code nearbyTextPattern} in
+ * format of {@code $x} and be picked from target file.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code ".*"}.
+ * </li>
+ * <li>
+ * Property {@code messagePattern} - Specify check violation message pattern to suppress.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code idPattern} - Specify check ID pattern to suppress.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code lineRange} - Specify negative/zero/positive value that
+ * defines the number of lines preceding/at/following the suppressing nearby text.
+ * Property can also be a RegExp group index at {@code nearbyTextPattern} in
+ * format of {@code $x} and be picked from target file.
+ * Type is {@code java.lang.String}.
+ * Default value is {@code "0"}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the filter to suppress audit events on the same line:
```

Comment:
All your examples are Java code, but https://github.com/checkstyle/checkstyle/issues/11939#issue-1311558875 has a non-Java example. Isn't this check made to support non-Java files? We should covnert some (more than 1) to non-Java examples.

Reply:
4 examples are Java and 4 are non-java. I'd say it is okay, no?

Reply:
Ok, maybe I just wasn't seeing the non-Java ones.